doc/memory

```diff
@@ -21,88 +22,85 @@ functions, including boot, filesystem access, context switching, etc.
 Volatile Flash access is generally only used with interrupts disabled to prevent
 other processes from ruining your day.
 
-                                   RAM Layout
+                                   RAM layout
 
 Right now, KnightOS only uses 0x8000 (32K) of RAM, even on devices that may have
 additional RAM available. Nothing of value is kept in RAM, and it may be safely
 wiped during a reboot (and in fact, the kernel does so).
 
-Note that RAM data structures are scheduled to change dramatically for kernel
-0.7.0.
-
     Address Length  Description
 
-    0x8000  0x100   Thread table
-    0x8100  0x40    Library table
-    0x8140  0x40    Signal table
-    0x8180  0x40    File handle table
-    0x8200  0x100   Misc kernel state
-    0x8300  0x100   Kernel flash functions
-    0x8400  0x200   Volatile kernel garbage
-    0x8600  0x7A00  Userspace memory
+    0x8000  0x80    Miscellaneous kernel state
+    0x8080  0x80    Flash execution space
+    0x8100  0x100   Volatile kernel utility space (kernelGarbage)
+    0x8200  0x7D00  Heap
+
+Misc kernel state holds several statically allocated kernel variables. Flash
+execution space is reserved for the Flash driver to use to execute Flash
+operations, which must be run from RAM. Volatile kernel utility space is a fixed
+buffer used for various kernel operations where dynamic memory allocation cannot
+fail. Finally, the heap is where `malloc` allocates memory from.
 
-Userspace memory is where `malloc` allocates memory. Volatile kernel garbage is
-used throughout the kernel whenever it just needs a little space to work, and is
-accessed with interrupts disabled. Kernel flash functions are where the Flash
-driver loads executable code to perform Flash manipulation from RAM. Misc kernel
-state is used for various things like the currently executing thread.
+                               The process table
 
-                                Data Structures
+The kernel supports preemptive multitasking, and to this end it keeps track of
+running processes and their resources. The process table is where this state is
+stored, and it is allocated in the heap and grows or shrinks and demands change.
+There are some variables in misc kernel state that keep track of the process
+table:
 
-Each of the tables is a fixed-length array that holds several of the following
-data structures.
+    process_table       Pointer to process table in the heap
+    active_processes    Number of active processes
+    current_process     Index of the currently running process
+    last_pid            The last PID allocated by the kernel
```

Comment:
These variables are currently called `threadTable`, `activeThreads`, `currentThreadIndex`, and `lastThreadID`. Part of this change is to rename all of the kernel internal memory to use snake_case. Also, note that `threadTable` is a fixed address currently - you'll have to stick two bytes of memory somewhere in the kernel state and update code that refers to threadTable to read that instead.